services/src/handlers/datasets.rs

```diff
@@ -869,6 +873,38 @@ fn column_map_to_column_vecs(columns: &HashMap<String, ColumnDataType>) -> Colum
     }
 }
 
+/// Delete a dataset
+#[utoipa::path(
+    tag = "Datasets",
+    delete,
+    path = "/dataset/{dataset}",
+    responses(
+        (status = 200, description = "OK"),
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+pub async fn delete_dataset_handler<C: Context>(
+    dataset: web::Path<DatasetId>,
+    session: AdminOrSession<C>,
+    ctx: web::Data<C>,
+) -> Result<impl Responder>
```

Comment:
Please define a type here

Reply:
why? We do it this way everywhere else.

Reply:
Some are created this way, e.g. https://github.com/geo-engine/geoengine/blob/5b3b5d4f30758594e18509e4f8d9faa9b0adf661/services/src/handlers/layers.rs#L835

What is the advantage of not specifying our returned type?

Reply:
convenience and uniformity across the code base.

Reply:
But it is our API shouldn't we at least specify the type here?

Reply:
0ca45624d8989b8c1a26f06879e31669e48e25f5